dashboard/src/components/Config/AppRepoList/AppRepoForm.tsx

```diff
@@ -0,0 +1,259 @@
+import * as React from "react";
+import { Redirect } from "react-router";
+
+interface IAppRepoFormProps {
+  message?: string;
+  redirectTo?: string;
+  install: (name: string, url: string, authHeader: string, customCA: string) => Promise<boolean>;
+  onAfterInstall?: () => Promise<any>;
+}
+
+interface IAppRepoFormState {
+  useRawAuthHeader: boolean;
+  useBasicAuth: boolean;
+  useBearerToken: boolean;
+  user: string;
+  password: string;
+  authHeader: string;
+  token: string;
+  name: string;
+  url: string;
+  customCA: string;
+}
+
+export class AppRepoForm extends React.Component<IAppRepoFormProps, IAppRepoFormState> {
+  public state = {
+    useRawAuthHeader: false,
+    useBasicAuth: false,
+    useBearerToken: false,
+    user: "",
+    password: "",
+    authHeader: "",
+    token: "",
+    name: "",
+    url: "",
+    customCA: "",
+  };
+
+  public render() {
+    return (
+      <form className="container padding-b-bigger" onSubmit={this.handleInstallClick}>
+        <div className="row">
+          <div className="col-12">
+            <div>
+              <h2>Add an App Repository</h2>
+            </div>
+            <div>
+              <label>
```

Comment:
Same comment about the usage of `htmlFor` and `id` on the `input`. Btw, you don't need to add the input inside the `label` for text and selectors:

`<label htmlFor="name">
  Name:
</label>
<input
  type="text"
  id="name"
  ...
/>
```